mod/01_core/set/all/fetch.rb

```diff
@@ -55,7 +55,9 @@ def fetch mark, opts={}
       if opts[:new]
         return card.renew(opts) if !clean_cache_opts? opts
       else
-        return unless !opts[:skip_virtual] && card.virtual?
+        card.include_set_modules unless opts[:skip_modules]  # need to load modules here to call the right virtual? method 
+        opts[:skip_modules] = true                           # don't load twice
+        return if opts[:skip_virtual] || !card.virtual?
```

Comment:
Right, that's what I meant.  The rule card just triggers the instantiating of a virtual? card, then it's content and type would never be used.  BTW, does a code-virtual card have a type?

Change is fine, but I don't think you need to set :skip_modules.  The loading of modules is always conditioned on them being already loaded.  There are lots of paths where it can get called multiple times, so that has to work or we'd be re-doing that step a lot.